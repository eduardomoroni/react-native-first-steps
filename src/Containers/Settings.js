// @flow

import React, { Component } from 'react'
import { Text, View } from 'react-native'
import styles from '../Styles/SettingsStyle'

export default class Settings extends Component {
  render () {
    return (
      <View style={styles.container}>
        <Text style={styles.instructions}>
          Settings
        </Text>
      </View>
    )
  }
}
