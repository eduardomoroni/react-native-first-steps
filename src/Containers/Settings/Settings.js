import React, { Component } from 'react'
import { View, Image, StyleSheet, ScrollView, ListView } from 'react-native'

import {
  List,
  ListItem,
  Text
} from 'react-native-elements'

const list1 = [
  {
    title: 'Appointments',
    icon: 'av-timer'
  },
  {
    title: 'Trips',
    icon: 'flight-takeoff'
  },
  {
    title: 'Passwords',
    icon: 'fingerprint'
  },
  {
    title: 'Pitches',
    icon: 'lightbulb-outline'
  },
  {
    title: 'Updates',
    icon: 'track-changes'
  }
]

class Settings extends Component {
  constructor () {
    super()
    const ds = new ListView.DataSource({rowHasChanged: (r1, r2) => r1 !== r2})
    this.state = {
      dataSource: ds.cloneWithRows(list1)
    }
    this.renderRow = this.renderRow.bind(this)
  }
  renderRow (rowData, sectionID) {
    return (
      <ListItem
        key={sectionID}
        title={rowData.title}
        icon={{name: rowData.icon}}
      />
    )
  }
  render () {
    return (
      <ScrollView keyboardShouldPersistTaps='always' style={styles.mainContainer}>
        <List>
          <ListView
            renderRow={this.renderRow}
            dataSource={this.state.dataSource}
            />
        </List>
        <List>
          <ListItem
            roundAvatar
            title='Limited supply! Its like digital gold!'
            subtitle={
              <View style={styles.subtitleView}>
                <Image source={require('../../Assets/Images/rating.png')} style={styles.ratingImage} />
                <Text style={styles.ratingText}>5 months ago</Text>
              </View>
            }
            avatar={require('../../Assets/Images/avatar1.jpg')}
          />
        </List>
      </ScrollView>
    )
  }
}

const styles = StyleSheet.create({
  mainContainer: {
    backgroundColor: '#ebedf1'
  },
  container: {
    marginTop: 60
  },
  heading: {
    color: 'white',
    marginTop: 10,
    fontSize: 22
  },
  hero: {
    marginTop: 60,
    justifyContent: 'center',
    alignItems: 'center',
    padding: 40,
    backgroundColor: '#69DDFF'
  },
  subtitleView: {
    flexDirection: 'row',
    paddingLeft: 10,
    paddingTop: 5
  },
  ratingImage: {
    height: 19.21,
    width: 100
  },
  ratingText: {
    paddingLeft: 10,
    color: 'grey'
  }
})

export default Settings
