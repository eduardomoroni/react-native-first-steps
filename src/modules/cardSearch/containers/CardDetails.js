// @flow

import React, { Component } from 'react'
import { Text, View, ScrollView } from 'react-native'
import { styles } from '../../theme/cardSearch/CardDetailsStyle'
import { CardImage } from '../../shared/components/index'
import { cardType, rulingsType } from '../../../types/CardType'
import { rulingsTextAsArray } from '../../../services/realm/conversion/RealmRepresentation'
import I18n from 'react-native-i18n'

export default class CardDetails extends Component {
  renderText (value: string, key: number) {
    return (
      <Text key={key} >{`
        - ${value}`
      }</Text>
    )
  }

  renderRulings (rulings: rulingsType) {
    const rulesText = rulingsTextAsArray(rulings)

    return (
      <View style={styles.rowContainer}>
        <Text style={styles.rowText}>
          <Text style={styles.rowName}>{`${I18n.t('rulings')}: `}</Text>
          { rulesText.map(this.renderText) }
        </Text>
      </View>
    )
  }

  renderRow (rowName: string, rowText: string) {
    if (rowText === null) {
      return null
    }

    return (
      <View style={styles.rowContainer}>
        <Text style={styles.rowText}>
          <Text style={styles.rowName}>{`${I18n.t(rowName)}: `}</Text>
          {rowText}
        </Text>
      </View>
    )
  }

  render () {
    const { card } = this.props

    return (
      <ScrollView style={styles.container}>
        <View style={styles.card}>
          <CardImage card={card} />
        </View>
        {this.renderRow('name', card.name)}
        {this.renderRow('type', card.type)}
        {this.renderRow('P/T', `${card.power}/${card.toughness}`)}
        {this.renderRow('manaCost', `${card.manaCost} (${card.cmc})`)}
        {this.renderRow('rarity', card.rarity)}
        {this.renderRow('text', card.text)}
        {this.renderRow('flavor', card.flavor)}
        {this.renderRulings(card.rulings)}
        {this.renderRow('sets', 'Estou com pregui√ßa')}
        {this.renderRow('legality', 'Farei depois')}
        {this.renderRow('foreignNames', 'Um dia')}
        <View style={styles.card} />
      </ScrollView>
    )
  }
}

CardDetails.propTypes = {
  card: cardType
}
