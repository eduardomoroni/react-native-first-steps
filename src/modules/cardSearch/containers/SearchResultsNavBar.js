// @flow

import React, { Component } from 'react'
import { connect } from 'react-redux'
import { navBackButton, navTitle, navButtonBar, openDrawer } from '../../modules/navigation/NavBarItems'
import { switchDisplayMode as switchDisplayModeAction } from '../../redux/actions'
import Menu from './ListCardFilterMenu'
import CustomNavBar from '../../modules/navigation/CustomNavBar'
import type { Dispatch } from 'redux'

export class SearchResultsNavBar extends Component {
  render () {
    const { showCardsAs, switchDisplayMode } = this.props

    const settingsButton = {
      buttonProps: {name: 'sliders'},
      callback: () => openDrawer(<Menu />)
    }

    const displayModeButton = {
      buttonProps: {name: showCardsAs === 'image' ? 'th-list' : 'th-large'},
      callback: () => switchDisplayMode(showCardsAs === 'image' ? 'list' : 'image')
    }

    return (
      <CustomNavBar
        leftRender={navBackButton()}
        middleRender={navTitle('searching_cards')}
        rightRender={navButtonBar([displayModeButton, settingsButton])} />
    )
  }
}

const mapStateToProps = (state) => {
  const { showCardsAs } = state.cardSearch
  return {
    showCardsAs
  }
}

const mapDispatchToProps = (dispatch: Dispatch<*>) => {
  return {
    switchDisplayMode: (toggle) => dispatch(switchDisplayModeAction(toggle))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(SearchResultsNavBar)
