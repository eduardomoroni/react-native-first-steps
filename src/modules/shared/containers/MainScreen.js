// @flow

import React, { Component } from 'react'
import {
  View,
  StyleSheet
} from 'react-native'
import { Colors } from '../modules/theme/themes'

export default class MainScreen extends Component {
  render () {
    return (
      <View style={styles.container} />
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    backgroundColor: Colors.background
  }
})
