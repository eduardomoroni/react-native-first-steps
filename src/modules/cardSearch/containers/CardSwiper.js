import React, { Component } from 'react'
import { View } from 'react-native'
import PropTypes from 'prop-types'
import CardDetails from './CardDetails'
import { Actions as NavigationActions } from 'react-native-router-flux'
import { cardType } from '../../../types/CardType'
import { findCardIndex } from '../../../services/CardService'
import Swiper from 'react-native-swiper'
import { FloatingActionButton } from '../../shared/components/index'

const fabItems = [
  { name: 'md-share',
    buttonColor: '#9b59b6',
    title: 'Share',
    onPress: () => null
  }, {
    name: 'md-folder-open',
    buttonColor: '#3498db',
    title: 'Add to Deck',
    onPress: () => null
  }, {
    name: 'md-swap',
    buttonColor: '#1abc9c',
    title: 'Add to TradeList',
    onPress: () => null
  }
]

export class CardSwiper extends Component {
  renderCardDetails (card, key) {
    return (
      <CardDetails card={card} key={key} />
    )
  }

  updateScreenTitle (e, state) {
    const { index } = state
    const getCardName = (index) => this.children[index].props.card.name
    NavigationActions.refresh({title: `${getCardName(index)}`})
  }

  render () {
    const { cards, card } = this.props
    let { index } = this.props

    if (card !== null) {
      index = findCardIndex(cards, card)
    }

    return (
      <View>
        <Swiper
          ref={(component) => { this.swiper = component }}
          onMomentumScrollEnd={this.updateScreenTitle}
          style={{ flex: 1 }}
          loadMinimal
          loadMinimalSize={2}
          loop={false}
          showsPagination={false}
          index={index}
          >
          {cards.map(this.renderCardDetails)}
        </Swiper>
        <FloatingActionButton items={fabItems} />
      </View>
    )
  }
}

CardSwiper.propTypes = {
  cards: PropTypes.objectOf(cardType),
  card: cardType,
  index: PropTypes.number
}

CardSwiper.defaultProps = {
  index: 0,
  card: null
}

export default CardSwiper
