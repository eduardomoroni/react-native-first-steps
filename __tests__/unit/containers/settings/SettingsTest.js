// @flow

import 'react-native'
import React from 'react'
import { shallow } from 'enzyme'
import { Settings } from '../../../../src/modules/settings/containers/Settings'

const props = {
  value: 'test'
}

describe('<settings />', () => {
  it('Snapshot', () => {
    const wrapper = shallow(<Settings {...props} />)
    expect(wrapper).toMatchSnapshot()
  })
})
