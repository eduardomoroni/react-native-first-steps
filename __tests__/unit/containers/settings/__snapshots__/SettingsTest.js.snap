// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<settings /> Snapshot 1`] = `
<ScrollViewMock
  keyboardShouldPersistTaps="always"
  style={
    Object {
      "backgroundColor": "#ebedf1",
    }
  }
>
  <List>
    <ListViewMock
      dataSource={
        ListViewDataSource {
          "items": 5,
        }
      }
      renderRow={[Function]}
      renderScrollComponent={[Function]}
    />
  </List>
  <List>
    <ListItem
      avatar={1}
      chevronColor="#bdc6cf"
      hideChevron={false}
      rightIcon={
        Object {
          "name": "chevron-right",
        }
      }
      roundAvatar={true}
      subtitle={
        <View
          style={
            Object {
              "flexDirection": "row",
              "paddingLeft": 10,
              "paddingTop": 5,
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": 19.21,
                "width": 100,
              }
            }
          />
          <TextElement
            style={
              Object {
                "color": "grey",
                "paddingLeft": 10,
              }
            }
          >
            5 months ago
          </TextElement>
        </View>
      }
      switchButton={false}
      textInputEditable={true}
      title="Limited supply! Its like digital gold!"
      underlayColor="white"
    />
  </List>
</ScrollViewMock>
`;
